tests/e2e/logging/test_openshift-logging.py

```diff
@@ -72,44 +72,82 @@ def finalizer():
-        elasticsearch_pod = [
-            pod.name for pod in pod_list if pod.name.startswith('elasticsearch')
-        ]
-        elasticsearch_pod_obj = get_pod_obj(
-            name=elasticsearch_pod[1], namespace='openshift-logging'
-        )
-        project_index = elasticsearch_pod_obj.exec_cmd_on_pod(
-            command='indices', out_yaml_format=False
-        )
-        project = pvc_obj.project.namespace
 
-        if project in project_index:
-            logger.info(f'The project {project} exists in the EFK stack')
-            for item in project_index.split("\n"):
-                if project in item:
-                    logger.info(item.strip())
-                    assert 'green' in item.strip(), f"Project {project} is Unhealthy"
+        elasticsearch_pod_obj = self.get_elasticsearch_pod_obj()
+        if get_ocp_version() <= "4.4":
+
+            project_index = elasticsearch_pod_obj.exec_cmd_on_pod(
+                command='indices', out_yaml_format=False
+            )
+            if project in project_index:
+                logger.info(f'The project {project} exists in the EFK stack')
+                for item in project_index.split("\n"):
+                    if project in item:
+                        logger.info(item.strip())
+                        assert 'green' in item.strip(), f"Project {project} is Unhealthy"
+            else:
+                raise ModuleNotFoundError
         else:
-            raise ModuleNotFoundError
+            cmd = (
+                'es_util --query=*/_search?pretty -d \'{"query": {"match":'
+                f' {{"kubernetes.namespace_name": "{project}"}}}}}}\''
+            )
+            project_out = elasticsearch_pod_obj.exec_cmd_on_pod(
+                command=cmd, out_yaml_format=False
+            )
+            if project_out:
+                logger.info("The Project exists on the EFK stack")
+            else:
+                raise ModuleNotFoundError
 
     def get_elasticsearch_pod_obj(self):
         """
         This function returns the Elasticsearch pod obj
+
         """
+
         pod_list = get_all_pods(namespace=constants.OPENSHIFT_LOGGING_NAMESPACE)
         elasticsearch_pod = [
-            pod for pod in pod_list if pod.name.startswith('elasticsearch')
+            pod for pod in pod_list if ('delete' not in pod.name) and (
+                'rollover' not in pod.name) and (
+                pod.name.startswith('elasticsearch')
+            )
         ]
         elasticsearch_pod_obj = random.choice(elasticsearch_pod)
         return elasticsearch_pod_obj
 
+    def check_filecount_in_project(self, project):
+        """
+        Check the files in the project
+
+        Args:
+            project (str): The project name
+
+        """
+
+        elasticsearch_pod_obj = self.get_elasticsearch_pod_obj()
+        cmd = f'es_util --query=project.{project}.*/_count'
+        if get_ocp_version() >= "4.5":
```

Comment:
is it intentional that the check here is if the cluster version is greater or equal to 4.5?

Reply:
yes it is, The logging commands and the way of checking for a project and files in EFK stack significantly changed from OCP 4.5. hence it is written in such a way that the test case should run on OCP 4.5 and versions lesser than that.